From 4ef3f87dacb4b21e7ce512af3ff9728b925a7265 Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Wed, 14 Oct 2020 19:40:18 +0300
Subject: [PATCH backport 4.19 1/1] mlxsw: minimal: Enforce interface MAC
 address setting to module Id

The purpose of this interface is only to provide ethtool interface
for reading module QSFP EEPROM data. Set MAC address up to the
module number just in order to have simple mapping between module
and this MAC address.

Signed-off-by: Vadim Pasternak <nvidia@mellanox.com>
---
 drivers/net/ethernet/mellanox/mlxsw/minimal.c | 17 +++++------------
 1 file changed, 5 insertions(+), 12 deletions(-)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/minimal.c b/drivers/net/ethernet/mellanox/mlxsw/minimal.c
index 8cc969758..2877423f5 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/minimal.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/minimal.c
@@ -27,7 +27,6 @@ struct mlxsw_m {
 	int *module_to_port;
 	struct mlxsw_core *core;
 	const struct mlxsw_bus_info *bus_info;
-	u8 base_mac[ETH_ALEN];
 	u8 max_ports;
 };
 
@@ -94,20 +93,14 @@ static const struct ethtool_ops mlxsw_m_port_ethtool_ops = {
 static int
 mlxsw_m_port_dev_addr_get(struct mlxsw_m_port *mlxsw_m_port)
 {
-	struct mlxsw_m *mlxsw_m = mlxsw_m_port->mlxsw_m;
 	struct net_device *dev = mlxsw_m_port->dev;
-	char ppad_pl[MLXSW_REG_PPAD_LEN];
-	int err;
 
-	mlxsw_reg_ppad_pack(ppad_pl, false, 0);
-	err = mlxsw_reg_query(mlxsw_m->core, MLXSW_REG(ppad), ppad_pl);
-	if (err)
-		return err;
-	mlxsw_reg_ppad_mac_memcpy_from(ppad_pl, dev->dev_addr);
-	/* The last byte value in base mac address is guaranteed
-	 * to be such it does not overflow when adding local_port
-	 * value.
+	/* The purpose of this interface is only to provide ethtool interface
+	 * for reading module QSFP EEPROM data. Set MAC address up to the
+	 * module number just in order to have simple mapping between module
+	 * and this MAC address.
 	 */
+	memset(dev->dev_addr, 0x00, ETH_ALEN - 1);
 	dev->dev_addr[ETH_ALEN - 1] = mlxsw_m_port->module + 1;
 	return 0;
 }
-- 
2.11.0

